api/middleware/middleware.go

```diff
@@ -49,7 +46,7 @@ func TokenMiddleware(nextHandler http.Handler) http.Handler {
 			visas, err := auth.GetVisas(auth.Details, token)
 			if err != nil {
 				log.Debug("failed to validate token at AAI")
-				http.Error(w, "bad token", 401)
+				c.String(http.StatusUnauthorized, "bad token")
```

Comment:
```suggestion
				c.String(http.StatusUnauthorized, "bad token")
				c.AbortWithError(code, err)
```
it looks like `c.String` only writes the response message, but then the request continues, so we need to abort it also